src/libraries/System.Runtime.Serialization.BinaryFormat/src/PayloadReader.cs

```diff
@@ -0,0 +1,285 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+
+namespace System.Runtime.Serialization.BinaryFormat;
+
+#if SYSTEM_RUNTIME_SERIALIZATION_BINARYFORMAT
+public
+#else
+internal
+#endif
+static class PayloadReader
+{
+    private static UTF8Encoding ThrowOnInvalidUtf8Encoding { get; } = new(false, throwOnInvalidBytes: true);
+
+    /// <summary>
+    /// Checks if given buffer contains only Binary Formatter payload.
+    /// </summary>
+    /// <param name="bytes">The buffer to inspect</param>
+    /// <returns>True if the first and last bytes indicate Binary Format, otherwise false.</returns>
+    public static bool ContainsBinaryFormatterPayload(ReadOnlySpan<byte> bytes)
+        => bytes.Length >= SerializedStreamHeaderRecord.Size + 2
+            && bytes[0] == (byte)RecordType.SerializedStreamHeader
+            && bytes[bytes.Length - 1] == (byte)RecordType.MessageEnd;
+
+    /// <summary>
+    /// Checks if given stream contains only Binary Formatter payload.
+    /// </summary>
+    /// <param name="stream">The readable and seekable stream to inspect.</param>
+    /// <returns>True if the first and last bytes indicate Binary Format, otherwise false.</returns>
```

Comment:
```suggestion
    /// <returns><see langword="true" /> if the first and last bytes indicate Binary Format; otherwise, <see langword="false" />.</returns>
```

https://github.com/dotnet/dotnet-api-docs/wiki/Returns:-Methods